src/app/api/attachments/attachments.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { NextRequest, NextResponse } from 'next/server'
+import authenticate from '@api/core/utils/authenticate'
+import { CreateAttachmentRequest, CreateAttachmentRequestSchema } from '@/types/dto/attachments.dto'
+import { AttachmentsService } from '@api/attachments/attachments.service'
+import httpStatus from 'http-status'
+import { handleUpload, type HandleUploadBody } from '@vercel/blob/client'
+import { IdParams } from '@api/core/types/api'
+
+export const createAttachment = async (request: NextRequest) => {
+  const user = await authenticate(request)
+  const body = (await request.json()) as HandleUploadBody
```

Comment:
Is it possible to avoid using a type cast here. The request body could be anything and we could be vulnerable to injecting manual data. If possible parse it using a schema instead so we throw an error if the request body does not match our expected schema.